src/sqlitestorage.cpp

```diff
@@ -570,28 +523,26 @@ bool SqliteStorage::loadNotebookIncidences(const QString &notebookUid)
 bool SqliteStorage::loadIncidenceInstance(const QString &instanceIdentifier)
 {
     QString uid;
-    QDateTime recId;
     // At the moment, from KCalendarCore, if the instance is an exception,
     // the instanceIdentifier will ends with yyyy-MM-ddTHH:mm:ss[Z|[+|-]HH:mm]
     // This is tested in tst_loadIncidenceInstance() to ensure that any
     // future breakage would be properly detected.
     if (instanceIdentifier.endsWith('Z')) {
         uid = instanceIdentifier.left(instanceIdentifier.length() - 20);
-        recId = QDateTime::fromString(instanceIdentifier.right(20), Qt::ISODate);
     } else if (instanceIdentifier.length() > 19
                && instanceIdentifier[instanceIdentifier.length() - 9] == 'T') {
         uid = instanceIdentifier.left(instanceIdentifier.length() - 19);
-        recId = QDateTime::fromString(instanceIdentifier.right(19), Qt::ISODate);
     } else if (instanceIdentifier.length() > 25
                && instanceIdentifier[instanceIdentifier.length() - 3] == ':') {
         uid = instanceIdentifier.left(instanceIdentifier.length() - 25);
-        recId = QDateTime::fromString(instanceIdentifier.right(25), Qt::ISODate);
-    }
-    if (!recId.isValid()) {
+    } else {
         uid = instanceIdentifier;
     }
 
-    return load(uid, recId);
+    // Even if we're looking for a specific incidence instance, we load all
+    // the series for recurring event, to avoid orphaned exceptions in the
+    // calendar or recurring events without their exceptions.
+    return load(uid);
```

Comment:
Wondering should we keep the loadIncidenceInstance() method or would the nemo-calendar code be simpler if it just loaded also full series. Can adjust separately, though, if so decided.

Reply:
As discussed, there could be some use when feeding events from buteo logs. Good enough reason as the implementation is quite simple and self-contained.

Reply:
Exactly, thank you for reminding the reason. The discussion was summarized with something like:

- instance identifiers can be used to identify unique occurrences (like for a sync log, or a search result),
- the (instance identifier -> UID) is not a trivial function,
- thus mKCal should provide a way to load a full event (parent + exception) knowing a given instance identifier.